release/build.sh

```diff
@@ -33,17 +33,55 @@ mkdir $output
 cp release/play_abstreet.$ext release/ungap_the_map.$ext release/INSTRUCTIONS.txt $output
 
 for name in game cli fifteen_min osm_viewer parking_mapper santa ltn; do
-	cp target/release/${name}${suffix} $output;
+    bin="target/release/${name}${suffix}"
+    if [[ "$output" = "abst_mac" ]]; then
+        # If this errors or hangs, ensure you've imported and unlocked a
+        # keychain holding a codesigning certificate with a Common Name
+        # matching "Developer ID Application.*"
+        codesign -fs "Developer ID Application" --timestamp -o runtime "$bin"
+    fi
+    cp "$bin" "$output";
 done
 
 mkdir $output/data
 cp -Rv data/system $output/data/system
 cp data/MANIFEST.json $output/data
 
-# Windows doesn't have zip?!
-if [[ "$output" != "abst_windows" ]]; then
-	# TODO Github will double-zip this, but if we just pass the directory, then the
-	# chmod +x bits get lost
-	zip -r $output $output
-	rm -rf release_data.zip $output
-fi
+case $os in
+    ubuntu-18.04)
+        # TODO Github will double-zip this, but if we just pass the directory, then the
+        # chmod +x bits get lost
+        zip -r $output $output
+        rm -rf $output
+        ;;
+
+    macos-latest)
+        ditto -c -k --keepParent $output $output.zip
+        xcrun notarytool submit --wait \
+            --apple-id $MACOS_DEVELOPER_APPLE_ID \
+            --team-id $MACOS_DEVELOPER_TEAM_ID \
+            --password $MACOS_DEVELOPER_APP_SPECIFIC_PASSWORD \
+            $output.zip
+
+        # TODO: staple the notarization so users can launch the app while
```

Comment:
Oh yeah! One other caveat about notarization (so many caveats!):

When macos launches an executable, it checks to see if it has been notarized. To do this it makes a network request to apples service to see if they have a record of the code with this signature as having been notarized. 

This gets cached by the users machine, so it's a one time cost, but if the user is offline, they can't check the notarization service, so they'll see a similar warning as now (though slightly different text - "Apple cannot check it for malicious software."). 

The solution here is called "stapling" - which embeds proof of notarization into the distributed application, so that the user can verify locally without having to make a network request. The *hitch* is that there's nowhere to store that notarization on  raw executable. Another thing we need a proper .app for.

(It's also possible to notarize .dmg and .pgks, but those formats have their own downsides, and ultimately I think we want a proper .app anyway).

I think that this PR doesn't hurt anything, and will result in a strictly better experience for the users who run the app from the CLI and are online the first time they do so. And it moves us closer to maybe one day having a signed proper .app